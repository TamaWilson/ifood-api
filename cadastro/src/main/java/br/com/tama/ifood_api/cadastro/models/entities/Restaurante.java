package br.com.tama.ifood_api.cadastro.models.entities;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name="restaurante")
public class Restaurante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String proprietario;

    public String cnpj;

    public String nome;

    @OneToOne(cascade = CascadeType.ALL)
    public Localizacao localizacao;

    @CreationTimestamp
    public LocalDateTime dataCriacao;

    @UpdateTimestamp
    public LocalDateTime dataAtualizacao;
}

