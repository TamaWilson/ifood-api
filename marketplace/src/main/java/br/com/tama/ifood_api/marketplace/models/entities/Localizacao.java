package br.com.tama.ifood_api.marketplace.models.entities;

public class Localizacao {

    public Long id;

    public Double latitude;

    public Double longitude;
}
