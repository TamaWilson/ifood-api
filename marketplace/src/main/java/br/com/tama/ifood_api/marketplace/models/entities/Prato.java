package br.com.tama.ifood_api.marketplace.models.entities;

import br.com.tama.ifood_api.marketplace.models.dto.PratoDTO;
import io.smallrye.mutiny.Multi;
import io.smallrye.mutiny.Uni;
import io.vertx.mutiny.pgclient.PgPool;
import io.vertx.mutiny.sqlclient.Row;
import io.vertx.mutiny.sqlclient.RowSet;
import io.vertx.mutiny.sqlclient.Tuple;

import java.math.BigDecimal;

import static br.com.tama.ifood_api.marketplace.utils.MutinyUtils.uniToMulti;

public class Prato {

    public Long id;

    public String nome;

    public String descricao;

    public Restaurante restaurante;

    public BigDecimal preco;

    public static Multi<PratoDTO> findAll(PgPool pgPool) {
        Uni<RowSet<Row>> preparedQuery = pgPool.preparedQuery("select * from prato").execute();
        return uniToMulti(preparedQuery);
    }

    public static Multi<PratoDTO> findAll(PgPool client, Long idRestaurante) {
        Uni<RowSet<Row>> preparedQuery = client
                .preparedQuery("SELECT * FROM prato where prato.restaurante_id = $1 ORDER BY nome ASC").execute(
                        Tuple.of(idRestaurante));
        return uniToMulti(preparedQuery);

    }

    public static Uni<PratoDTO> findById(PgPool client, Long id) {
        return client.preparedQuery("SELECT * FROM prato WHERE id = $1").execute(Tuple.of(id))
                .map(RowSet::iterator)
                .map(iterator -> iterator.hasNext() ? PratoDTO.from(iterator.next()) : null);
    }

}
