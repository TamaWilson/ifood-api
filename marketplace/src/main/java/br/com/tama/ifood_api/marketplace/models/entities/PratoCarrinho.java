package br.com.tama.ifood_api.marketplace.models.entities;

public class PratoCarrinho {

    public String usuario;

    public Long prato;
}
