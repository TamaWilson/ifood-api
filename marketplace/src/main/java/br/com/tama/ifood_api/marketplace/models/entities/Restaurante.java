package br.com.tama.ifood_api.marketplace.models.entities;

public class Restaurante {

    public Long id;

    public String nome;

    public Localizacao localizacao;
}
