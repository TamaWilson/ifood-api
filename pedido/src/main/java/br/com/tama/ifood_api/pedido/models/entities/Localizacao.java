package br.com.tama.ifood_api.pedido.models.entities;

public class Localizacao {

    public Double latitude;

    public Double longitude;

}
