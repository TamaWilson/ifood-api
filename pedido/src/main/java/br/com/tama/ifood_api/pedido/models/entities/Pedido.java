package br.com.tama.ifood_api.pedido.models.entities;

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import io.quarkus.mongodb.panache.common.MongoEntity;

import java.util.List;

@MongoEntity(collection = "pedido", database = "pedido")
public class Pedido  extends PanacheMongoEntity {

    public String cliente;

    public Restaurante restaurante;

    public List<Prato> pratos;

    public String entregador;

    public Localizacao localizacao;
}
