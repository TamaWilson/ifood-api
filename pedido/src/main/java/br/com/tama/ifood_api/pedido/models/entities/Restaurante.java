package br.com.tama.ifood_api.pedido.models.entities;

public class Restaurante {

    public String nome;

}
