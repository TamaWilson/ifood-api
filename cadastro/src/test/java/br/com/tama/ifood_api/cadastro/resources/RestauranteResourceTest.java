package br.com.tama.ifood_api.cadastro.resources;

import br.com.tama.ifood_api.cadastro.config.CadastroTestLifecycleManager;
import br.com.tama.ifood_api.cadastro.models.entities.Restaurante;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.quarkus.test.security.TestSecurity;
import io.restassured.http.ContentType;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
@TestSecurity(user = "proprietario", roles = "proprietario")
class RestauranteResourceTest {


    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void testBuscarRestaurantes(){
        String resultado = given()
                .contentType(ContentType.JSON)
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
       Approvals.verifyJson(resultado);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void testAlterarRestaurante(){
        Restaurante dto = new Restaurante();
        dto.nome = "novoNome";
        Long parameterValue = 123L;
        given()
                .contentType(ContentType.JSON)
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();

        Restaurante findById = Restaurante.findById(parameterValue);
    }
}
