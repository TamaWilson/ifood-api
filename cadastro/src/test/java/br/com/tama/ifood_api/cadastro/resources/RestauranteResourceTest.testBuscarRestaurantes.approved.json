[
  {
    "id": 123,
    "nomeFantasia": "Novo Restaurante",
    "proprietario": "ID DO KEYCLOAK"
  },
  {
    "id": 1234,
    "nomeFantasia": "Novo Restaurante 2",
    "proprietario": "ID DO KEYCLOAK 2"
  }
]